presto-main/src/main/java/io/prestosql/execution/buffer/PagesSerde.java

```diff
@@ -80,7 +78,7 @@ public Page deserialize(SerializedPage serializedPage)
     {
         checkArgument(serializedPage != null, "serializedPage is null");
 
-        if (!decompressor.isPresent() || serializedPage.getCompression() == UNCOMPRESSED) {
+        if (!decompressor.isPresent() || !COMPRESSED.isSet(serializedPage.getPageCodecMarkers())) {
```

Comment:
Maybe for this we just add an `isCompressed` method to `SerializedPage`.  If we need more complex stuff in the future, we can just change this since it is an internal detail of Presto.

Reply:
Yep, will mark this as `isCompressed` and do the same for `isEncrypted`.